components/director/internal/tenantmapping/mapper_for_system_auth.go

```diff
@@ -24,25 +24,39 @@ type mapperForSystemAuth struct {
 	scopesGetter  ScopesGetter
 }
 
-func (m *mapperForSystemAuth) GetTenantAndScopes(ctx context.Context, reqData ReqData, authID string, authFlow AuthFlow) (string, string, error) {
+func (m *mapperForSystemAuth) GetObjectContext(ctx context.Context, reqData ReqData, authID string, authFlow AuthFlow) (ObjectContext, error) {
 	sysAuth, err := m.systemAuthSvc.GetGlobal(ctx, authID)
 	if err != nil {
-		return "", "", errors.Wrap(err, "while retrieving system auth from database")
+		return ObjectContext{}, errors.Wrap(err, "while retrieving system auth from database")
 	}
 
 	refObjType, err := sysAuth.GetReferenceObjectType()
 	if err != nil {
-		return "", "", errors.Wrap(err, "while getting reference object type")
+		return ObjectContext{}, errors.Wrap(err, "while getting reference object type")
 	}
 
+	var scopes string
+	var tenant string
+
 	switch refObjType {
 	case model.IntegrationSystemReference:
-		return m.getTenantAndScopesForIntegrationSystem(reqData)
+		tenant, scopes, err = m.getTenantAndScopesForIntegrationSystem(reqData)
```

Comment:
I would move this switch to private function and put `GetReferenceObjectType` and `GetReferenceObjectID` closer to each other.